### PR TITLE
Sagemaker Data Quality, Model Quality, Model Explainability Job Definitions + Compilation, Processing Jobs

### LINKED ISSUE #9462: Add support for SageMaker Model Quality Job Definition resource
### Describe the feature

Would be useful to support SageMaker Model Quality Job Definitions as a aws.sagemaker-model-quality-job-definition resource type

### Extra information or context

_No response_

### PR DESCRIPTION
**Pull request by mattheidelbaugh**
Closes #9462 , #9461, #9437, #9440, #9421

### CODE REVIEW

File: c7n/resources/sagemaker.py [hunk 1]
```diff
@@ -220,6 +220,82 @@ def resources(self, query=None):
...
+    source_mapping = {'describe': SagemakerProcessingJobDescribe}
+
+    def __init__(self, ctx, data):
+        super(SagemakerProcessingJob, self).__init__(ctx, data)
+        self.queries = QueryFilter.parse(
+            self.data.get('query', [
+                {'StatusEquals': 'InProgress'}]))
+
+    def resources(self, query=None):
+        for q in self.queries:
```
Comment (kapilt) [inline c7n/resources/sagemaker.py:290]: this block feels duplicated a lot, feels like its ripe for being extracted to a utility function.

Comment (kapilt) [reply c7n/resources/sagemaker.py:290]: separate pr though in that context.

Comment (kapilt) [reply c7n/resources/sagemaker.py:290]: filed as https://github.com/cloud-custodian/cloud-custodian/issues/9469

**Review by PratMis — APPROVED** [review]
lgtm, thanks

**Pull request merged**